sql/sql_select.cc

```diff
@@ -1895,7 +1897,27 @@ int JOIN::init_join_caches()
     {
       table->prepare_for_keyread(tab->index, table->read_set);
     }
-    if (tab->cache && tab->cache->init(select_options & SELECT_DESCRIBE))
+
+    /*
+      We are about to initialize a join buffer.
+      Can we use lightweight initalization mode just for EXPLAINs? We can if
+      we're certain that the optimizer will not execute the subquery.
+
+      It won't execute it if this join is too expensive.  We follow the logic
+      in Item_subselect::is_expensive() and initalize for explain only if this
+      join examines more than @@expensive_subquery_limit.
+    */
+    if (tab->cache && !checked_if_expensive)
```

Comment:
`checked_if_expensive` is set once during the loop and is never reset, so it affects other tables. Is this the idea?